articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+
+annotation pos as "22:DiagramPosition"
+
+car *- pos: <<annotations>>
+
+annotation review as "33:ReviewComment"
+
+steering *- review: <<annotations>>
+
+engine -[hidden]> steering
+----
+
+In summary, a node's parent must always be symmetric -- either to the parent's children, or the parent's annotation instances.
+
+If we deleted the parent node, we implicitly delete all its children and annotation instances, each of their children and annotation instances, and so on.
+In other words, a child or annotation instance shares the lifetime of its parent.
+
+This also explains why we consider the annotated node the parent of an annotation instance:
+If the annotated node is gone, the annotation instance makes no sense anymore -- they share the annotated node's lifetime.
+
+[[properties]]
+=== Properties
+
+We use properties to store simple values inside a node.
+A node of classifier `Person` might have properties `name: String`, `age: Integer` and `alive: Boolean`.
+
+[plantuml, personProperties, svg]
+----
+hide empty members
+class person as "3:Person" {
+  name: String = "Joe"
+  age: Integer = 23
+  alive: Boolean = true
+}
+----
+
+A node's classifier describes which properties can appear in the node.
+This description includes the property's _name_, _key_, whether it is _optional_, and its _type_.
+The name is for humans to understand the property; we use the key to uniquely identify it technically.
+With the optional flag we specify whether a node without this property is considered valid or not:
+In our system, a `person` without known `name` might be a problem, but we could still process them without knowing their `age`.
+
+A property's type describes what kind of values can appear in that property.
+All LionWeb data types, and thus all properties, are _value typesfootnote:[https://en.wikipedia.org/w/index.php?title=Value_type]_.
+This means they don't have any identity, we just know their value: We can distinguish `10` and `15`, but `12` and `12` are the same.
+
+Some modeling frameworks allow properties with multiple values, e.g. `luckyNumbers: Integer[0..*] = {23, 42, 4711}`.
+LionWeb does not support that, because the values don't have identity.
+So if we first had a property value of `{23, 42}`, and then `{23, 42, 42}`, we have no way of telling whether the new `42` has been inserted or appended.
+This leads to all kinds of problems we'd like to avoid.
+
+The following subsections describe the kinds of data types supported by LionWeb.
+
+==== Primitive types
+LionWeb ships with built-in types for `String`, `Integer`, and `Boolean`, as they are needed in almost all systems.
+String properties can have any string value, including empty.
+Integer properties support positive and negative decimal integers of any length.
+Boolean properties can only have the values `true` or `false`.
+A language might define its own additional _primitive types_.
+In any case, we store all of them as string on low-level serialization format (i.e. JSON).
```

Comment:
"_on_ low-level serialization format" doesn't sound entirely right to me

Reply:
maybe "_in_ low-level serialization format"? I try to avoid calling it directly JSON, as that might be confused with the JSON type we used to have.

Reply:
Then maybe "_in **the**_ low-level serialization format"?